docs/content/docs/concepts/summarization.md

```diff
@@ -0,0 +1,101 @@
+
+### Summary Lifecycle
+
+The lifecycle of a summary starts when a "parent summarizer" client is elected.
+
+-   The parent summarizer spawns a non-interactive summarizer client.
+-   The summarizer client periodically starts a summary as per heuristics. A summary happens at a particular sequence number called the "summary sequence number" or reference sequence number for the summary.
+-   The container runtime (or simply runtime) generates a summary tree (described in the ["What does a summary look like?"](#what-does-a-summary-look-like) section below).
+-   The runtime uploads the summary tree to the Fluid storage service which returns a handle (unique id) to the summary if the upload is successful. Otherwise, it returns a failure.
+-   The runtime submits a "summarize" op to the Fluid ordering service containing the uploaded summary handle and the summary sequence number.
+-   The ordering service stamps it with a sequence number (like any other op) and broadcasts the summarize op.
+-   Another service on the server responds to the summarize op.
+    -   If the summary is accepted, it sends a "summary ack" with the summary sequence number and a summary handle. This handle may or may not be the same as the one in summary op depending on whether this is a single-commit or two-commit summary. More details on this below.
+    -   If the summary is rejected, it sends a "summary nack" with the details of the summary op
+-   The runtime completes the summary process on receiving the summary ack / nack. The runtime has a timeout called "maxAckWaitTime" and if the summary op, ack or nack is not received within this time, it will fail this summary.
+
+### Single-commit vs two-commit summaries
+
+By default, Fluid uses "two-commit summaries" mode where the two commits refer to the storage committing the summary twice and returning two different handles for it - One when the summary is uploaded and second, on responding to the summary op via a summary ack. In this mode, when the server receives the summary op, it augments the corresponding summary with a "protocol" blob hence generating a new commit and new handle for this summary which it returns in the summary ack.
+
+Fluid is switching to "single-commit summary" mode where the client adds the "protocol" blob when uploading the summary. Thus, the server doesn't need to augment the summary and the summary ack is no longer required. As soon as the summary is uploaded (first commit), the summary process is complete. The "summarize" op then is just a way to indicate that a summary happened, and it has details of the summary
```

Comment:
That's a good point. The problem is that we are in the middle of switching to single commit summaries. One of the services supports it and it's enabled for them but for others we are still using two-commnit summaries.

Maybe I can say that there are 2 summary modes and it can be configured which mode to use. We are actively working on switching to single-commit summaries by FF 3.0?

Reply:
> That's a good point. The problem is that we are in the middle of switching to single commit summaries. One of the services supports it and it's enabled for them but for others we are still using two-commnit summaries.
> 
> Maybe I can say that there are 2 summary modes and it can be configured which mode to use. We are actively working on switching to single-commit summaries by FF 3.0?

Do Fluid app devs need to know the implementation details of summaries? How does this help them create Fluid apps?

Reply:
Not really. But this document is a bit more of a deep dive into the concept of summarization. However, I removed this section. This document will be updated once we completely move to single-commit summaries.